make.py

```diff
@@ -0,0 +1,249 @@
+"""
+Helper script to build langkit and its dependencies.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+from contextlib import contextmanager
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+
+
+from langkit.utils import add_search_path
+
+
+parser = argparse.ArgumentParser(
+    description='Helper script to build langkit and its dependencies'
+)
+parser.add_argument('--build-mode', default='dev')
+
+subparsers = parser.add_subparsers()
+
+build_parser = subparsers.add_parser('build')
+build_parser.set_defaults(command='build')
```

Comment:
I suggest you move the parsers declaration after the function definitions and add `func=XXX` (`func=build`, `install` and `setenv`) to all subparsers? This will make manual dispatching starting line 242 unecessary. See similar processing in `libmanage.py`.

Reply:
Ah this is nice!